src/jcs.rs

```diff
@@ -78,8 +56,31 @@ impl Ord for JsonProperty {
 
 type JsonObject = BTreeSet<JsonProperty>;
 
+/// An RFC 8785 compatible JSON Canonicalization Scheme (JCS) formatter for [serde_json].
+///
+/// # Example
+/// ```
+/// use serde::Serialize;
+/// use serde_json::Serializer;
+/// use serde_json_canonicalizer::JcsFormatter;
+///
+/// #[derive(Serialize)]
+/// struct Data {
+///     c: isize,
+///     b: bool,
+///     a: String,
+/// }
+///
+/// let data = Data { c: 120, b: false, a: "Hello!".to_owned() };
+///
+/// let mut ser = Serializer::with_formatter(Vec::new(), JcsFormatter::default());
+/// data.serialize(&mut ser).unwrap();
+/// let result = ser.into_inner();
+///
+/// assert_eq!(&result, r#"{"a":"Hello!","b":false,"c":120}"#.as_bytes());
+/// ```
 #[derive(Default)]
-struct JcsFormatter {
+pub struct JcsFormatter {
```

Comment:
The `JcsFormatter` itself is not RFC 8785 compatible because the `Serializer` handles non finite floating point numbers. 
(Check https://github.com/serde-rs/json/issues/1020)
So using the formatter on its own without the custom serializer will not produce the expected result.

That is why it is not exposed for consumers of the library.

Reply:
Ah I've overlooked that. Fully agree and have removed the `pub` from `JcsFormatter` and replaced the docstring. Thanks!

Reply:
Actually there are issues exporting the JcsSerializer type because it exposes the JcsFormatter in the interface (although we can make `new()` non-pub as a workaround).

Is there a clear reason or serde convention why the serializer should return the io error, or could we move this logic to the formatter?

Reply:
> Actually there are issues exporting the JcsSerializer type because it exposes the JcsFormatter in the interface (although we could make JcsFormatter::new() non-pub as a workaround and hide it in the docs).

Yes, so I had to make `JcsSerializer` internal also or I could expose `JcsFormatter` and risk accidental misuse of the library. I think exposing only the utility functions is fine. 

> Is there a clear reason or serde convention why the serializer should return the io error, or could we move this logic to the formatter?

Unfortunately the default `Serializer` will write `null` for infinite floats and not let the formatter handle the number.
`JcsSerializer` only exists because of this design choice in `serde_json`. Otherwise having the formatter only would be sufficient.
Although there is no comment on https://github.com/serde-rs/json/issues/1020, I think the problem with changing the serializer is that now anyone using `serde_json` with a custom formatter relies on not having to deal with non-finite floats in the formatter.

Reply:
I see the original serde_json Serializer is the problem now. Okay, have moved both to `pub(crate)` instead to prevent exposing them in the public interface.